A config flag to run errcheck with the verbose flag

### Welcome

- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Docker

### Your feature request related to a problem? Please describe

There currently seems to be no way to run the errcheck linter with its verbose flag enabled. This is sometimes frustrating when trying to update the exclude list, because by default, there is no indication from the linter as to what specific package/type/function triggered the warning.

The difference is that without verbose mode the output looks like this:

```
Error return value of `io.Close` is not checked (errcheck)
        defer io.Close(ctx)
```

Which give a good hint, but it's still obtuse. was it triggered with a pointer receiver? Is it an interface and I need to exclude all possible things which satisfy the interface? Which package is it in? Which type is it? etc... 

With verbose mode (running errcheck manually), i get this:

```
(*github.com/username/project/internal/mypackage.MyType).Close      defer io.Close(ctx)
```

Where that first bit is **precisely** what I'd want to add to my exclude list. I'd prefer not to have run errcheck manually to get this information.

### Describe the solution you'd like

My proposal is to add:

```
  settings:
    errcheck:
        verbose: true
```

as an option to the `.golangci.yaml` file, and have it give the verbose output of the tool.

### Describe alternatives you've considered

I currently manually re-run errcheck directly from the command line to get the information I need from the tool :-(.

### Additional context

_No response_

### Supporter

- [ ] I am a sponsor through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

Hello,

The position (line, column) is inside the reports:


```console
$ golangci-lint run   
foo.go:4:5: Error return value is not checked (errcheck)
        foo()
           ^
1 issues:
* errcheck: 1
``

--- a/.golangci.next.reference.yml
+++ b/.golangci.next.reference.yml
@@ -408,6 +408,10 @@ linters:
         - io.Copy(*bytes.Buffer)
         - io.Copy(os.Stdout)
 
+      # Display function signature instead of selector.
+      # Default: false
+      verbose: true
+
     errchkjson:
       # With check-error-free-encoding set to true, errchkjson does warn about errors
       # from json encoding functions that are safe to be ignored,

--- a/jsonschema/golangci.next.jsonschema.json
+++ b/jsonschema/golangci.next.jsonschema.json
@@ -1115,6 +1115,11 @@
               "description": "To disable the errcheck built-in exclude list",
               "type": "boolean",
               "default": false
+            },
+            "verbose": {
+              "description": "Display function signature instead of selector",
+              "type": "boolean",
+              "default": false
             }
           }
         },

--- a/pkg/config/linters_settings.go
+++ b/pkg/config/linters_settings.go
@@ -372,6 +372,7 @@ type ErrcheckSettings struct {
 	CheckTypeAssertions      bool     `mapstructure:"check-type-assertions"`
 	CheckAssignToBlank       bool     `mapstructure:"check-blank"`
 	ExcludeFunctions         []string `mapstructure:"exclude-functions"`
+	Verbose                  bool     `mapstructure:"verbose"`
 }
 
 type ErrChkJSONSettings struct {

--- a/pkg/golinters/errcheck/errcheck.go
+++ b/pkg/golinters/errcheck/errcheck.go
@@ -37,7 +37,7 @@ func New(settings *config.ErrcheckSettings) *goanalysis.Linter {
 			checker.Tags = lintCtx.Cfg.Run.BuildTags
 
 			analyzer.Run = func(pass *analysis.Pass) (any, error) {
-				issues := runErrCheck(pass, checker)
+				issues := runErrCheck(pass, checker, settings.Verbose)
 
 				if len(issues) == 0 {
 					return nil, nil
@@ -56,7 +56,7 @@ func New(settings *config.ErrcheckSettings) *goanalysis.Linter {
 		WithLoadMode(goanalysis.LoadModeTypesInfo)
 }
 
-func runErrCheck(pass *analysis.Pass, checker *errcheck.Checker) []goanalysis.Issue {
+func runErrCheck(pass *analysis.Pass, checker *errcheck.Checker, verbose bool) []goanalysis.Issue {
 	pkg := &packages.Package{
 		Fset:      pass.Fset,
 		Syntax:    pass.Files,
@@ -76,6 +76,9 @@ func runErrCheck(pass *analysis.Pass, checker *errcheck.Checker) []goanalysis.Is
 
 		if err.FuncName != "" {
 			code := cmp.Or(err.SelectorName, err.FuncName)
+			if verbose {
+				code = err.FuncName
+			}
 
 			text = fmt.Sprintf("Error return value of %s is not checked", internal.FormatCode(code))
 		}

